simplesource-data/src/main/java/io/simplesource/data/CommandError.java

```diff
@@ -0,0 +1,158 @@
+package io.simplesource.data;
+
+import java.util.Objects;
+import java.util.function.BiFunction;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * A CommandError explains failures. They can be constructed from the reason type,
+ * with either a {@link String} or a {@link Throwable} for more context.
+ */
+public abstract class CommandError {
```

Comment:
Should this be CommandError, if it's in the data package, should it not be Error or something similar, CommandError seems to tie it to the command API

Reply:
Yes, it is tied to CommandAPI, this is why I will move it to simplesource-api module.
We had a discussion regarding the name, either use global name like Error or keep CommandError as it is. The only error type we have is CommandError.